core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -858,9 +858,9 @@ private void loadMetadata(LockedInodePath inodePath, LoadMetadataContext context
    */
   public static List<Journal.JournalEntry> mergeCreateComplete(
       List<alluxio.proto.journal.Journal.JournalEntry> entries) {
-    List<alluxio.proto.journal.Journal.JournalEntry> newEntries = new ArrayList<>();
+    List<alluxio.proto.journal.Journal.JournalEntry> newEntries = new ArrayList<>(entries.size());
     // file id : index in the newEntries, InodeFileEntry
-    Map<Long, Pair<Integer, MutableInodeFile>> fileEntryMap = new HashMap<>();
+    Map<Long, Pair<Integer, MutableInodeFile>> fileEntryMap = new HashMap<>(entries.size());
```

Comment:
Init with the incoming list size is not optimal because the Update commands will be merged into existing Create entries so the total size needed will be smaller. 

The default size of ArrayList is 10 and HashMap is 16. For now I think making a guess with the `entries.size()` will be overall smaller than the default sizes and still acceptable.